notification/slack/usergroupsender.go

```diff
@@ -0,0 +1,159 @@
+package slack
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/slack-go/slack"
+	"github.com/target/goalert/config"
+	"github.com/target/goalert/notification"
+	"github.com/target/goalert/permission"
+	"github.com/target/goalert/user"
+	"github.com/target/goalert/util/log"
+)
+
+// UserGroupSender processes on-call notifications by updating the members of a Slack user group.
+type UserGroupSender struct {
+	*ChannelSender
+}
+
+var _ notification.Sender = (*UserGroupSender)(nil)
+
+// UserGroupSender returns a new UserGroupSender wrapping the given ChannelSender.
+func (s *ChannelSender) UserGroupSender() *UserGroupSender {
+	return &UserGroupSender{s}
+}
+
+// Send implements notification.Sender.
+func (s *UserGroupSender) Send(ctx context.Context, msg notification.Message) (*notification.SentMessage, error) {
+	err := permission.LimitCheckAny(ctx, permission.User, permission.System)
+	if err != nil {
+		return nil, err
+	}
+
+	t, ok := msg.(notification.ScheduleOnCallUsers)
+	if !ok {
+		return nil, errors.Errorf("unsupported message type: %T", msg)
+	}
+
+	if t.Dest.Type != notification.DestTypeSlackUG {
+		return nil, errors.Errorf("unsupported destination type: %s", t.Dest.Type.String())
+	}
+
+	teamID, err := s.TeamID(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("lookup team ID: %w", err)
+	}
+
+	var userIDs []string
+	for _, u := range t.Users {
+		userIDs = append(userIDs, u.ID)
+	}
+
+	userSlackIDs := make(map[string]string, len(t.Users))
+	err = s.cfg.UserStore.AuthSubjectsFunc(ctx, "slack:"+teamID, userIDs, func(sub user.AuthSubject) error {
```

Comment:
To promote good secure coding habits, we should never directly concatenate strings.
```suggestion
	err = s.cfg.UserStore.AuthSubjectsFunc(ctx, fmt.Sprintf("slack:%s", teamID), userIDs, func(sub user.AuthSubject) error {
```